Ability to ignore a folder if sync files are present

Currently if syncthing is still downloading files a .tmp will be preset for each item still being downloaded. This will cause a failed extraction as the split rar's are not all complete. Retries handle this to some extent but on a slow connection unless retry is a high number or 0 they will run out of attempts.

Is it possible to check the directory for any files ending in .tmp (or user defined extension) and then skip over that folder until the next scan? I couldn't see an option to supply any extensions to cause this behaviour but may have missed it.

Thanks!

Forced an error, it knows of the syncthing key just nothing showing in logs mentioning syncthing

`[ERROR] Using Config File: /config/unpackerr.conf: config file: unmarshaling file [/config/unpackerr.conf]: toml: line 66 (last key "sonarr.syncthing"): incompatible types: TOML value has type string; destination has type boolean`

--- a/README.md
+++ b/README.md
@@ -110,6 +110,7 @@ sonarr.protocols|`UN_SONARR_0_PROTOCOLS`|`torrent` Protocols to process. Alt: `t
 sonarr.timeout|`UN_SONARR_0_TIMEOUT`|`10s` / How long to wait for the app to respond|
 sonarr.delete_orig|`UN_SONARR_0_DELETE_ORIG`|`false` / Delete archives after import? Recommend not setting this to true|
 sonarr.delete_delay|`UN_SONARR_0_DELETE_DELAY`|`5m` / Extracts are deleted this long after import, `-1s` to disable|
+sonarr.syncthing|`UN_SONARR_0_SYNCTHING`|`false` / Setting this to true makes unpackerr wait for syncthing to finish|
 
 ##### Radarr
 
@@ -122,6 +123,7 @@ radarr.protocols|`UN_RADARR_0_PROTOCOLS`|`torrent` Protocols to process. Alt: `t
 radarr.timeout|`UN_RADARR_0_TIMEOUT`|`10s` / How long to wait for the app to respond|
 radarr.delete_orig|`UN_RADARR_0_DELETE_ORIG`|`false` / Delete archives after import? Recommend not setting this to true|
 radarr.delete_delay|`UN_RADARR_0_DELETE_DELAY`|`5m` / Extracts are deleted this long after import, `-1s` to disable|
+radarr.syncthing|`UN_RADARR_0_SYNCTHING`|`false` / Setting this to true makes unpackerr wait for syncthing to finish|
 
 ##### Lidarr
 
@@ -134,6 +136,7 @@ lidarr.protocols|`UN_LIDARR_0_PROTOCOLS`|`torrent` Protocols to process. Alt: `t
 lidarr.timeout|`UN_LIDARR_0_TIMEOUT`|`10s` / How long to wait for the app to respond|
 lidarr.delete_orig|`UN_LIDARR_0_DELETE_ORIG`|`false` / Delete archives after import? Recommend not setting this to true|
 lidarr.delete_delay|`UN_LIDARR_0_DELETE_DELAY`|`5m` / Extracts are deleted this long after import, `-1s` to disable|
+lidarr.syncthing|`UN_LIDARR_0_SYNCTHING`|`false` / Setting this to true makes unpackerr wait for syncthing to finish|
 
 ##### Readarr
 
@@ -146,6 +149,7 @@ readarr.protocols|`UN_READARR_0_PROTOCOLS`|`torrent` Protocols to process. Alt: 
 readarr.timeout|`UN_READARR_0_TIMEOUT`|`10s` / How long to wait for the app to respond|
 readarr.delete_orig|`UN_READARR_0_DELETE_ORIG`|`false` / Delete archives after import? Recommend not setting this to true|
 readarr.delete_delay|`UN_READARR_0_DELETE_DELAY`|`5m` / Extracts are deleted this long after import, `-1s` to disable|
+readarr.syncthing|`UN_READARR_0_SYNCTHING`|`false` / Setting this to true makes unpackerr wait for syncthing to finish|
 
 ##### Folder
 

--- a/examples/unpackerr.conf.example
+++ b/examples/unpackerr.conf.example
@@ -72,6 +72,8 @@ dir_mode = "0755"
 ## General recommendation is: do not enable this for torrent use.
 ## Setting this to true deletes the entire original download folder after import.
 #  delete_orig = false
+## If you use Syncthing, setting this to true will make unpackerr wait for syncs to finish.
+# syncthing = false
 
 #[[radarr]]
 #  url = "http://127.0.0.1:7878"
@@ -86,6 +88,8 @@ dir_mode = "0755"
 ## General recommendation is: do not enable this for torrent use.
 ## Setting this to true deletes the entire original download folder after import.
 #  delete_orig = false
+## If you use Syncthing, setting this to true will make unpackerr wait for syncs to finish.
+# syncthing = false
 
 #[[lidarr]]
 #  url = "http://127.0.0.1:8686"
@@ -100,6 +104,8 @@ dir_mode = "0755"
 ## General recommendation is: do not enable this for torrent use.
 ## Setting this to true deletes the entire original download folder after import.
 #  delete_orig = false
+## If you use Syncthing, setting this to true will make unpackerr wait for syncs to finish.
+# syncthing = false
 
 #[[readarr]]
 #  url = "http://127.0.0.1:8787"
@@ -114,6 +120,8 @@ dir_mode = "0755"
 ## General recommendation is: do not enable this for torrent use.
 ## Setting this to true deletes the entire original download folder after import.
 #  delete_orig = false
+## If you use Syncthing, setting this to true will make unpackerr wait for syncs to finish.
+# syncthing = false
 
 
 ##################################################################################
@@ -143,6 +151,7 @@ dir_mode = "0755"
 #  move_back = false
 ## Set this to true if you want this app to extract ISO files with .iso extension.
 # extract_isos = false
+
 
 ################
 ### Webhooks ###

--- a/pkg/unpackerr/handlers.go
+++ b/pkg/unpackerr/handlers.go
@@ -6,12 +6,14 @@ import (
 	"strings"
 	"time"
 
+	"golift.io/cnfg"
 	"golift.io/starr"
 	"golift.io/xtractr"
 )
 
 // Extract holds data for files being extracted.
 type Extract struct {
+	Syncthing   bool
 	Retries     uint
 	Path        string
 	App         string
@@ -21,6 +23,16 @@ type Extract struct {
 	Status      ExtractStatus
 	IDs         map[string]interface{}
 	Resp        *xtractr.Response
+}
+
+// Shared config items for all starr apps.
+type starrConfig struct {
+	Path        string        `json:"path" toml:"path" xml:"path" yaml:"path"`
+	Paths       []string      `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
+	Protocols   string        `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`
+	DeleteOrig  bool          `json:"delete_orig" toml:"delete_orig" xml:"delete_orig" yaml:"delete_orig"`
+	DeleteDelay cnfg.Duration `json:"delete_delay" toml:"delete_delay" xml:"delete_delay" yaml:"delete_delay"`
+	Syncthing   bool          `json:"syncthing" toml:"syncthing" xml:"syncthing" yaml:"syncthing"`
 }
 
 // checkQueueChanges checks each item for state changes from the app queues.
@@ -85,6 +97,13 @@ func (u *Unpackerr) handleCompletedDownload(name string, x *Extract) {
 			item.App, name, item.Path, err)
 
 		return
+	}
+
+	if x.Syncthing {
+		if tmpFile := u.hasSyncThingFile(item.Path); tmpFile != "" {
+			u.Printf("[%s] Completed item still syncing: %s, found Syncthing .tmp file: %s", item.App, name, tmpFile)
+			return
+		}
 	}
 
 	item.Status = QUEUED
@@ -222,4 +241,15 @@ func (u *Unpackerr) isComplete(status, protocol, protos string) bool {
 	}
 
 	return false
+}
+
+func (u *Unpackerr) hasSyncThingFile(dirPath string) string {
+	files, _ := u.Xtractr.GetFileList(dirPath)
+	for _, file := range files {
+		if strings.HasPrefix(file, ".tmp") {
+			return file
+		}
+	}
+
+	return ""
 }

--- a/pkg/unpackerr/handlers.go
+++ b/pkg/unpackerr/handlers.go
@@ -246,7 +246,7 @@ func (u *Unpackerr) isComplete(status, protocol, protos string) bool {
 func (u *Unpackerr) hasSyncThingFile(dirPath string) string {
 	files, _ := u.Xtractr.GetFileList(dirPath)
 	for _, file := range files {
-		if strings.HasPrefix(file, ".tmp") {
+		if strings.HasSuffix(file, ".tmp") {
 			return file
 		}
 	}

--- a/pkg/unpackerr/handlers.go
+++ b/pkg/unpackerr/handlers.go
@@ -26,7 +26,7 @@ type Extract struct {
 }
 
 // Shared config items for all starr apps.
-type starrConfig struct {
+type StarrConfig struct {
 	Path        string        `json:"path" toml:"path" xml:"path" yaml:"path"`
 	Paths       []string      `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
 	Protocols   string        `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`

--- a/pkg/unpackerr/handlers.go
+++ b/pkg/unpackerr/handlers.go
@@ -243,6 +243,7 @@ func (u *Unpackerr) isComplete(status, protocol, protos string) bool {
 	return false
 }
 
+// added for https://github.com/davidnewhall/unpackerr/issues/235
 func (u *Unpackerr) hasSyncThingFile(dirPath string) string {
 	files, _ := u.Xtractr.GetFileList(dirPath)
 	for _, file := range files {

--- a/pkg/unpackerr/lidarr.go
+++ b/pkg/unpackerr/lidarr.go
@@ -12,7 +12,7 @@ import (
 // LidarrConfig represents the input data for a Lidarr server.
 type LidarrConfig struct {
 	starr.Config
-	starrConfig
+	StarrConfig
 	Queue          *lidarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*lidarr.Lidarr `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -75,18 +75,18 @@ func (u *Unpackerr) validateLidarr() error {
 func (u *Unpackerr) logLidarr() {
 	if c := len(u.Lidarr); c == 1 {
 		u.Printf(" => Lidarr Config: 1 server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-			"delete_orig: %v, delete_delay: %v, paths:%q",
+			"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 			u.Lidarr[0].URL, u.Lidarr[0].APIKey != "", u.Lidarr[0].Timeout,
-			u.Lidarr[0].ValidSSL, u.Lidarr[0].Protocols, u.Lidarr[0].DeleteOrig,
-			u.Lidarr[0].DeleteDelay.Duration, u.Lidarr[0].Paths)
+			u.Lidarr[0].ValidSSL, u.Lidarr[0].Protocols, u.Sonarr[0].Syncthing,
+			u.Lidarr[0].DeleteOrig, u.Lidarr[0].DeleteDelay.Duration, u.Lidarr[0].Paths)
 	} else {
 		u.Print(" => Lidarr Config:", c, "servers")
 
 		for _, f := range u.Lidarr {
 			u.Printf(" =>    Server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-				"delete_orig: %v, delete_delay: %v, paths:%q",
+				"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 				f.URL, f.APIKey != "", f.Timeout, f.ValidSSL, f.Protocols,
-				f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
+				f.Syncthing, f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
 		}
 	}
 }

--- a/pkg/unpackerr/lidarr.go
+++ b/pkg/unpackerr/lidarr.go
@@ -5,7 +5,6 @@ import (
 	"strings"
 	"sync"
 
-	"golift.io/cnfg"
 	"golift.io/starr"
 	"golift.io/starr/lidarr"
 )
@@ -13,11 +12,7 @@ import (
 // LidarrConfig represents the input data for a Lidarr server.
 type LidarrConfig struct {
 	starr.Config
-	Path           string        `json:"path" toml:"path" xml:"path" yaml:"path"`
-	Paths          []string      `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
-	Protocols      string        `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`
-	DeleteOrig     bool          `json:"delete_orig" toml:"delete_orig" xml:"delete_orig" yaml:"delete_orig"`
-	DeleteDelay    cnfg.Duration `json:"delete_delay" toml:"delete_delay" xml:"delete_delay" yaml:"delete_delay"`
+	starrConfig
 	Queue          *lidarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*lidarr.Lidarr `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -141,6 +136,7 @@ func (u *Unpackerr) checkLidarrQueue() {
 					App:         Lidarr,
 					DeleteOrig:  server.DeleteOrig,
 					DeleteDelay: server.DeleteDelay.Duration,
+					Syncthing:   server.Syncthing,
 					Path:        u.getDownloadPath(q.StatusMessages, Lidarr, q.Title, server.Paths),
 					IDs: map[string]interface{}{
 						"title":      q.Title,

--- a/pkg/unpackerr/radarr.go
+++ b/pkg/unpackerr/radarr.go
@@ -12,7 +12,7 @@ import (
 // RadarrConfig represents the input data for a Radarr server.
 type RadarrConfig struct {
 	starr.Config
-	starrConfig
+	StarrConfig
 	Queue          *radarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*radarr.Radarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -75,18 +75,18 @@ func (u *Unpackerr) validateRadarr() error {
 func (u *Unpackerr) logRadarr() {
 	if c := len(u.Radarr); c == 1 {
 		u.Printf(" => Radarr Config: 1 server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-			"delete_orig: %v, delete_delay: %v, paths:%q",
+			"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 			u.Radarr[0].URL, u.Radarr[0].APIKey != "", u.Radarr[0].Timeout,
-			u.Radarr[0].ValidSSL, u.Radarr[0].Protocols, u.Radarr[0].DeleteOrig,
-			u.Radarr[0].DeleteDelay.Duration, u.Radarr[0].Paths)
+			u.Radarr[0].ValidSSL, u.Radarr[0].Protocols, u.Sonarr[0].Syncthing,
+			u.Radarr[0].DeleteOrig, u.Radarr[0].DeleteDelay.Duration, u.Radarr[0].Paths)
 	} else {
 		u.Print(" => Radarr Config:", c, "servers")
 
 		for _, f := range u.Radarr {
 			u.Printf(" =>    Server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-				"delete_orig: %v, delete_delay: %v, paths:%q",
+				"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 				f.URL, f.APIKey != "", f.Timeout, f.ValidSSL, f.Protocols,
-				f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
+				f.Syncthing, f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
 		}
 	}
 }

--- a/pkg/unpackerr/radarr.go
+++ b/pkg/unpackerr/radarr.go
@@ -5,7 +5,6 @@ import (
 	"strings"
 	"sync"
 
-	"golift.io/cnfg"
 	"golift.io/starr"
 	"golift.io/starr/radarr"
 )
@@ -13,11 +12,7 @@ import (
 // RadarrConfig represents the input data for a Radarr server.
 type RadarrConfig struct {
 	starr.Config
-	Path           string        `json:"path" toml:"path" xml:"path" yaml:"path"`
-	Paths          []string      `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
-	Protocols      string        `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`
-	DeleteOrig     bool          `json:"delete_orig" toml:"delete_orig" xml:"delete_orig" yaml:"delete_orig"`
-	DeleteDelay    cnfg.Duration `json:"delete_delay" toml:"delete_delay" xml:"delete_delay" yaml:"delete_delay"`
+	starrConfig
 	Queue          *radarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*radarr.Radarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -141,6 +136,7 @@ func (u *Unpackerr) checkRadarrQueue() {
 					App:         Radarr,
 					DeleteOrig:  server.DeleteOrig,
 					DeleteDelay: server.DeleteDelay.Duration,
+					Syncthing:   server.Syncthing,
 					Path:        u.getDownloadPath(q.StatusMessages, Radarr, q.Title, server.Paths),
 					IDs: map[string]interface{}{
 						"downloadId": q.DownloadID,

--- a/pkg/unpackerr/readarr.go
+++ b/pkg/unpackerr/readarr.go
@@ -5,7 +5,6 @@ import (
 	"strings"
 	"sync"
 
-	"golift.io/cnfg"
 	"golift.io/starr"
 	"golift.io/starr/readarr"
 )
@@ -13,11 +12,7 @@ import (
 // ReadarrConfig represents the input data for a Readarr server.
 type ReadarrConfig struct {
 	starr.Config
-	Path             string         `json:"path" toml:"path" xml:"path" yaml:"path"`
-	Paths            []string       `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
-	Protocols        string         `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`
-	DeleteOrig       bool           `json:"delete_orig" toml:"delete_orig" xml:"delete_orig" yaml:"delete_orig"`
-	DeleteDelay      cnfg.Duration  `json:"delete_delay" toml:"delete_delay" xml:"delete_delay" yaml:"delete_delay"`
+	starrConfig
 	Queue            *readarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex     `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*readarr.Readarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -141,6 +136,7 @@ func (u *Unpackerr) checkReadarrQueue() {
 					App:         Readarr,
 					DeleteOrig:  server.DeleteOrig,
 					DeleteDelay: server.DeleteDelay.Duration,
+					Syncthing:   server.Syncthing,
 					Path:        u.getDownloadPath(q.StatusMessages, Readarr, q.Title, server.Paths),
 					IDs: map[string]interface{}{
 						"title":      q.Title,

--- a/pkg/unpackerr/readarr.go
+++ b/pkg/unpackerr/readarr.go
@@ -12,7 +12,7 @@ import (
 // ReadarrConfig represents the input data for a Readarr server.
 type ReadarrConfig struct {
 	starr.Config
-	starrConfig
+	StarrConfig
 	Queue            *readarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex     `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*readarr.Readarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -75,18 +75,18 @@ func (u *Unpackerr) validateReadarr() error {
 func (u *Unpackerr) logReadarr() {
 	if c := len(u.Readarr); c == 1 {
 		u.Printf(" => Readarr Config: 1 server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-			"delete_orig: %v, delete_delay: %v, paths:%q",
+			"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 			u.Readarr[0].URL, u.Readarr[0].APIKey != "", u.Readarr[0].Timeout,
-			u.Readarr[0].ValidSSL, u.Readarr[0].Protocols, u.Readarr[0].DeleteOrig,
-			u.Readarr[0].DeleteDelay.Duration, u.Readarr[0].Paths)
+			u.Readarr[0].ValidSSL, u.Readarr[0].Protocols, u.Sonarr[0].Syncthing,
+			u.Readarr[0].DeleteOrig, u.Readarr[0].DeleteDelay.Duration, u.Readarr[0].Paths)
 	} else {
 		u.Print(" => Readarr Config:", c, "servers")
 
 		for _, f := range u.Readarr {
 			u.Printf(" =>    Server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-				"delete_orig: %v, delete_delay: %v, paths:%q",
+				"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 				f.URL, f.APIKey != "", f.Timeout, f.ValidSSL, f.Protocols,
-				f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
+				f.Syncthing, f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
 		}
 	}
 }

--- a/pkg/unpackerr/sonarr.go
+++ b/pkg/unpackerr/sonarr.go
@@ -5,7 +5,6 @@ import (
 	"strings"
 	"sync"
 
-	"golift.io/cnfg"
 	"golift.io/starr"
 	"golift.io/starr/sonarr"
 )
@@ -13,11 +12,7 @@ import (
 // SonarrConfig represents the input data for a Sonarr server.
 type SonarrConfig struct {
 	starr.Config
-	Path           string        `json:"path" toml:"path" xml:"path" yaml:"path"`
-	Paths          []string      `json:"paths" toml:"paths" xml:"paths" yaml:"paths"`
-	Protocols      string        `json:"protocols" toml:"protocols" xml:"protocols" yaml:"protocols"`
-	DeleteOrig     bool          `json:"delete_orig" toml:"delete_orig" xml:"delete_orig" yaml:"delete_orig"`
-	DeleteDelay    cnfg.Duration `json:"delete_delay" toml:"delete_delay" xml:"delete_delay" yaml:"delete_delay"`
+	starrConfig
 	Queue          *sonarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*sonarr.Sonarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -141,6 +136,7 @@ func (u *Unpackerr) checkSonarrQueue() {
 					App:         Sonarr,
 					DeleteOrig:  server.DeleteOrig,
 					DeleteDelay: server.DeleteDelay.Duration,
+					Syncthing:   server.Syncthing,
 					Path:        u.getDownloadPath(q.StatusMessages, Sonarr, q.Title, server.Paths),
 					IDs: map[string]interface{}{
 						"title":      q.Title,

--- a/pkg/unpackerr/sonarr.go
+++ b/pkg/unpackerr/sonarr.go
@@ -12,7 +12,7 @@ import (
 // SonarrConfig represents the input data for a Sonarr server.
 type SonarrConfig struct {
 	starr.Config
-	starrConfig
+	StarrConfig
 	Queue          *sonarr.Queue `json:"-" toml:"-" xml:"-" yaml:"-"`
 	sync.RWMutex   `json:"-" toml:"-" xml:"-" yaml:"-"`
 	*sonarr.Sonarr `json:"-" toml:"-" xml:"-" yaml:"-"`
@@ -75,18 +75,18 @@ func (u *Unpackerr) validateSonarr() error {
 func (u *Unpackerr) logSonarr() {
 	if c := len(u.Sonarr); c == 1 {
 		u.Printf(" => Sonarr Config: 1 server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-			"delete_orig: %v, delete_delay: %v, paths:%q",
+			"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 			u.Sonarr[0].URL, u.Sonarr[0].APIKey != "", u.Sonarr[0].Timeout,
-			u.Sonarr[0].ValidSSL, u.Sonarr[0].Protocols, u.Sonarr[0].DeleteOrig,
-			u.Sonarr[0].DeleteDelay.Duration, u.Sonarr[0].Paths)
+			u.Sonarr[0].ValidSSL, u.Sonarr[0].Protocols, u.Sonarr[0].Syncthing,
+			u.Sonarr[0].DeleteOrig, u.Sonarr[0].DeleteDelay.Duration, u.Sonarr[0].Paths)
 	} else {
 		u.Print(" => Sonarr Config:", c, "servers")
 
 		for _, f := range u.Sonarr {
 			u.Printf(" =>    Server: %s, apikey:%v, timeout:%v, verify ssl:%v, protos:%s, "+
-				"delete_orig: %v, delete_delay: %v, paths:%q",
+				"syncthing: %v, delete_orig: %v, delete_delay: %v, paths:%q",
 				f.URL, f.APIKey != "", f.Timeout, f.ValidSSL, f.Protocols,
-				f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
+				f.Syncthing, f.DeleteOrig, f.DeleteDelay.Duration, f.Paths)
 		}
 	}
 }

